src/extension/script.ts

```diff
@@ -0,0 +1,147 @@
+import { fetchStream, streamBody } from "./fetchStream";
+
+import { getCurrentTab } from "./utils";
+
+async function cherryPickCommits(path, commitBranch, targetBranch) {
+  const table = document.querySelector(".table");
+  const jsonFormdata = {};
+  jsonFormdata.commits = [];
+  jsonFormdata.localPath = path;
+  jsonFormdata.commitBranch = commitBranch;
+  jsonFormdata.targetBranch = targetBranch;
+  Array.from(table.rows).forEach((element, rowNumber) => {
+    if (rowNumber > 0) {
+      jsonFormdata.commits.push({
+        commitID: rowNumber,
+        commitScope: element.cells[0].firstChild.checked,
+        commitSHA: element.cells[1].firstChild.value,
+      });
+    }
+  });
+  disableButton();
+  try {
+    await fetchStream(
+      `http://localhost:4000/cherrypick`,
+      "POST",
+      jsonFormdata,
+      (chunkString) => {
+        setContentInDesc(chunkString);
+      }
+    ).then((res) => {
+      enableButton();
+      setContentInDesc(res);
+    });
+  } catch (e) {
+    enableButton();
+  }
+}
+function setContentInDesc(content) {
+  const el = document.getElementById("cherry-pick-desc");
+  el.style.display = "block";
+  el.textContent = content;
+}
+function disableButton() {
+  const buttons = document.querySelectorAll("button");
+  buttons.forEach((button) => {
+    button.setAttribute("disabled", "true");
+  });
+}
+function enableButton() {
+  const buttons = document.querySelectorAll("button");
+  buttons.forEach((button) => {
+    button.removeAttribute("disabled");
+  });
+}
+function renderFormElement(commit) {
```

Comment:
which element ? Choose function and variables name carefully. Currently they do not convey what they are doing or that they represent. Check this everywhere.